src/Azure.php

```diff
@@ -62,11 +64,65 @@ public function handle($request, Closure $next)
         return $this->handlecallback($request, $next, $access_token, $refresh_token);
     }
 
+    /**
+     * Handle an incoming request in a testing environment
+     * Assumes tester is calling actingAs or loginAs during testing to run this correctly
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param Closure $next
+     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector|mixed
+     */
+    protected function handleTesting(Request $request, Closure $next)
+    {
+        $user = Auth::user();
+
+        if (!isset($user))
+        {
+            return $this->redirect($request, $next);
+        }
+
+        return $this->handlecallback($request, $next);
+    }
+
+    /**
+     * Gets the azure url
+     *
+     * @return String
+     */
+    public function getAzureUrl()
+    {
+        return $this->baseUrl . env('AZURE_TENANT_ID') . $this->route . "authorize?response_type=code&client_id=" . env('AZURE_CLIENT_ID') . "&resource=" . urlencode(env('AZURE_RESOURCE'));
+    }
+
+    /**
+     * Redirects to the Azure route
```

Comment:
Can you add explanation of when this hits? Like "Redirects to the Azure route after blah blah is successful"